### ISSUE TITLE
AndroidManifest not parsed correctly

### ISSUE THREAD

**Issue by mingyuan-xia** [openstf/adbkit-apkreader#12]
[AndroidManifest.xml.zip](https://github.com/openstf/adbkit-apkreader/files/1930693/AndroidManifest.xml.zip)

apkreader 3.0.0 failed to parse this AXML. Only top-level info is extracted (versionCode) and the inner parts (`permissions`) are missing. No exception is thrown, though.

I am attaching the parsed result from `jadx`:
```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="27" android:versionName="1.2.9" package="com.vip.pinganedai">
    <uses-sdk android:minSdkVersion="16" android:targetSdkVersion="25"/>
    <permission android:name="com.vip.pinganedai.permission.JPUSH_MESSAGE" android:protectionLevel="signature"/>
    <uses-permission android:name="com.vip.pinganedai.permission.JPUSH_MESSAGE"/>
    <uses-permission android:name="android.permission.READ_CALL_LOG"/>
    <uses-permission android:name="android.permission.READ_SMS"/>
    <uses-permission android:name="android.permission.RECEIVE_USER_PRESENT"/>
    <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.BLUETOOTH"/>
    <uses-permission android:name="android.permission.WAKE_LOCK"/>
    <uses-permission android:name="android.permission.READ_PHONE_STATE"/>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.VIBRATE"/>
    <uses-permission android:name="android.permission.MOUNT_UNMOUNT_FILESYSTEMS"/>
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
    <uses-permission android:name="android.permission.WRITE_SETTINGS"/>
    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
    <uses-permission android:name="android.permission.READ_CONTACTS"/>
    <uses-permission android:name="android.permission.READ_CALENDAR"/>
    <uses-permission android:name="android.permission.WRITE_CALENDAR"/>
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
    <uses-permission android:name="android.permission.ACCESS_GPS"/>
    <uses-permission android:name="android.permission.ACCESS_ASSISTED_GPS"/>
    <uses-permission android:name="android.permission.ACCESS_LOCATION"/>
    <uses-permission android:name="android.permission.RECORD_AUDIO"/>
    <uses-permission android:name="android.permission.SEND_SMS"/>
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
    <uses-permission android:name="android.permission.ACCESS_COARSE_UPDATES"/>
    <uses-permission android:name="android.permission.CHANGE_WIFI_STATE"/>
    <uses-permission android:name="android.permission.ACCESS_LOCATION_EXTRA_COMMANDS"/>
    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE"/>
    <uses-permission android:name="android.permission.GET_TASKS"/>
    <uses-permission android:name="android.permission.CHANGE_WIFI_MULTICAST_STATE"/>
    <uses-permission android:name="android.permission.READ_LOGS"/>
    <uses-permission android:name="android.permission.CALL_PHONE"/>
    <uses-permission android:name="android.permission.SET_DEBUG_APP"/>
    <uses-permission android:name="android.permission.GET_ACCOUNTS"/>
    <uses-permission android:name="android.permission.USE_CREDENTIALS"/>
    <uses-permission android:name="android.permission.MANAGE_ACCOUNTS"/>
    <uses-permission android:name="android.permission.CAMERA"/>
    <uses-feature android:name="android.hardware.camera"/>
    <uses-feature android:name="android.hardware.camera.autofocus"/>
    <uses-permission android:name="android.permission.FLAG_GRANT_READ_URI_PERMISSION"/>
    <application android:theme="@style/AppTheme" android:label="中鸿贷" android:icon="@mipmap/applogo" android:name="com.vip.pinganedai.app.AppApplication" android:debuggable="true" android:allowBackup="true" android:hardwareAccelerated="true" android:supportsRtl="true">
        <meta-data android:name="UMENG_CHANNEL" android:value="Alpha"/>
        <meta-data android:name="UMENG_APPKEY" android:value="5aa73d8cb27b0a52570000f4"/>
        <activity android:theme="@style/Theme_AppLauncher" android:name="com.vip.pinganedai.ui.main.activity.SplashActivity" android:screenOrientation="portrait">
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
        <activity android:theme="@style/dispatcher" android:name="com.vip.pinganedai.ui.main.scheme.ui.DispatcherActivity">
            <intent-filter>
                <action android:name="android.intent.action.VIEW"/>
                <category android:name="android.intent.category.DEFAULT"/>
                <category android:name="android.intent.category.BROWSABLE"/>
                <data android:scheme="flashloan" android:host="www.xulu.com"/>
            </intent-filter>
        </activity>
        <activity android:name="com.vip.pinganedai.ui.main.activity.MainActivity" android:launchMode="singleTask" android:screenOrientation="portrait" android:configChanges="orientation" android:windowSoftInputMode="adjustUnspecified|stateHidden|adjustResize"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.CheckLoginActivity" android:launchMode="singleTop" android:screenOrientation="portrait" android:alwaysRetainTaskState="true"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.LoginActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.SmsActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.RegistActivity" android:launchMode="singleTask" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.main.activity.GuideActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.PhoneOperatorActivity" android:screenOrientation="portrait" android:configChanges="keyboardHidden|orientation|screenSize" android:windowSoftInputMode="adjustResize"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.FlashSettingsActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.FeedBackActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.LoanRecordsActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.UrgentLinkmanActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.main.activity.GuideH5Activity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.main.activity.H5ZhanShiActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.UserLoadH5Activity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.KeFuActivity" android:launchMode="singleTop" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.repayment.activity.CardPayStep2Activity" android:launchMode="singleTop" android:screenOrientation="portrait"/>
        <meta-data android:name="PARTNER_CODE" android:value="paed"/>
        <meta-data android:name="com.amap.api.v2.apikey" android:value="31b1545949994a55965d5d2405d0120c"/>
        <meta-data android:name="com.taobao.android.hotfix.IDSECRET" android:value="24779881-1"/>
        <meta-data android:name="com.taobao.android.hotfix.APPSECRET" android:value="1d8c622982666d37b871eab8dd31aa16"/>
        <meta-data android:name="com.taobao.android.hotfix.RSASECRET" android:value="MIIEvwIBADANBgkqhkiG9w0BAQEFAASCBKkwggSlAgEAAoIBAQDWptCZbC7u2I9+JkB0iSylG8mOxelibDe1BWezGPNwBdzsrRdNxbbtEacOq3mD7fyFvfNlw7Pau8G5lbKsrclMNuIkoD2DKMvSFzXuvvEXLFVQMbBF1v01OveOHyrvFsw6iWWdpAHTbkGpdcFnqnIemfwoaXKbGclXfuFR1AVrRV1vL+ebxXsNF7pwePsOALXzIgQPXcL+ShzOglg6/oFgdVj85Mhggnqr1ZpL/H9YRv/ef4sZu2LKfHXips3/vpeZTiqeYU1CFmmmSnsw3GSdKS3MM7Ig0s4tufv4lxwNOY1YQn0d1o1bwH1UNsMMLJstiQDXey83NB4jE/Zy92iNAgMBAAECggEBAI+u2EFqvjvZMKwztL+NNxpKcHAmoKPmZp/Av3qOQB4Dfq/a6ZeAuoMUZuwV8mBShM+NqAi9qH3JYtL0lsaqCe3/av8zWBmczFqUeOCYZc+qOG1W5XTKhXIFR8KVsGqcqxaf9CpCp4o86q0hitX2rS84yibO9w1FhJYR84eP0hTL/rt7sEfEh+v09PjRkG9cebCu91jGHJMR8yW5SC1z/3q08pu+3G4nBSXJZ9mRFNFRs+RrtCnNf+A48R2mZfMMUqZ6rCt2Utp88ik8OVs1RJWzY+Ux1/Q4WBqpCWBjQ0uAAeVHKhcIwStjxyjBqw3Ont3osyZn9tM7jGzwai1N4iECgYEA8/mLnUKBmBjEgeoO63jpqzRx9GxCAsCQ4KuzTRhIneAJjTGM29STqHYg8JQw2FDuaEPrXperYJ7exFNqcjwaSPEfzglL6o/i8dqSdLgHyll/nuZN+uBpmNhQ2Z9Kk99DYLXseXsc3Oe3XRAI3ht3mh3vpFcJAzqHzON3Do4voEUCgYEA4TtFppgw6r30tuI3VLX30kpwi/WXYAp5AG2wKA7i1yHKCdh3Ak0uN4QjYJND7UZX++wRoQqI6VGAYS7V0YLvgwbBjc3yufBSW4t3letKkFCbEiRhUV/E/nBD3L1l+hFQKAQ7oSdVW3yaQMvi2aPNWhvBzqaCudemcWZJPCYpX6kCgYEAm2d8D+q6A7jbCumdIU5+gQhCAfAnR/uDDSkxZ80tDCsVynMihzBr8aV1K+kpB9DdW1rQBQvVJRFtUl7T8fnowEee+62eRsXqSCXynhhPFgxWJ1AJoeDqQmqfbU9ywKg5EdQv0d2ir++FM+kvzNJwJ6Ij8GlkG0NgFxlYz9Ov8IUCgYBlEDKVqU6x1mCbO3B2e1xjKPHNa3rsJmbbMVzM7LkOrWO5/T6Rcs5dLNau/AWLz3elW+0hJNgAUYjYkqpw417TnVF2bsYOdu6Nk0wqojSIAxNCFb7xQ5biUL+yqe27Qs+D1TpoenzLm2Kf1pzI6b44Q6I4NvELsN9HnQ0Zh0ojWQKBgQDqi5nM6OuhXc/ZcnO126cgjye2gH0yVkgQvNxYoe/zFNLXrRYQHJQFDd4xAdZZ0yidqRH6j6wsDZw/RqAB7wsZwono2nFnQLrH/2+amWumHtSmomU6Oo/rCkvE/DQUuvZ1mplzlldSRO+kNdnX8n2zbKYHAGwOoxjHA4cZXGItqQ=="/>
        <service android:name="com.amap.api.location.APSService"/>
        <provider android:name="com.vip.pinganedai.utils.update.UpdateFileProvider" android:exported="false" android:authorities="com.vip.pinganedai.updatefileprovider" android:grantUriPermissions="true">
            <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/update_cache_path"/>
        </provider>
        <provider android:name="android.support.v4.content.FileProvider" android:exported="false" android:authorities="com.vip.pinganedai.fileprovider" android:grantUriPermissions="true">
            <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/file_paths"/>
        </provider>
        <activity android:name="com.megvii.livenesslib.LivenessActivity" android:screenOrientation="portrait"/>
        <activity android:theme="@android:style/Theme.Holo.NoActionBar.Fullscreen" android:label="@string/app_name" android:name="com.megvii.idcardlib.IDCardScanActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.megvii.demo.BankCardScanActivity" android:launchMode="singleTop" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.ConfirmBankActivity" android:launchMode="singleTop" android:screenOrientation="portrait"/>
        <service android:name="cn.jpush.android.service.PushService" android:enabled="true" android:exported="false">
            <intent-filter>
                <action android:name="cn.jpush.android.intent.REGISTER"/>
                <action android:name="cn.jpush.android.intent.REPORT"/>
                <action android:name="cn.jpush.android.intent.PushService"/>
                <action android:name="cn.jpush.android.intent.PUSH_TIME"/>
            </intent-filter>
        </service>
        <provider android:name="cn.jpush.android.service.DataProvider" android:exported="false" android:authorities="com.vip.pinganedai.DataProvider"/>
        <service android:name="cn.jpush.android.service.DaemonService" android:enabled="true" android:exported="true">
            <intent-filter>
                <action android:name="cn.jpush.android.intent.DaemonService"/>
                <category android:name="com.vip.pinganedai"/>
            </intent-filter>
        </service>
        <provider android:name="cn.jpush.android.service.DownloadProvider" android:exported="true" android:authorities="com.vip.pinganedai.DownloadProvider"/>
        <receiver android:name="cn.jpush.android.service.PushReceiver" android:enabled="true">
            <intent-filter android:priority="1000">
                <action android:name="cn.jpush.android.intent.NOTIFICATION_RECEIVED_PROXY"/>
                <category android:name="com.vip.pinganedai"/>
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.USER_PRESENT"/>
                <action android:name="android.net.conn.CONNECTIVITY_CHANGE"/>
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.PACKAGE_ADDED"/>
                <action android:name="android.intent.action.PACKAGE_REMOVED"/>
                <data android:scheme="package"/>
            </intent-filter>
        </receiver>
        <activity android:theme="@android:style/Theme.NoTitleBar" android:name="cn.jpush.android.ui.PushActivity" android:exported="false" android:configChanges="keyboardHidden|orientation">
            <intent-filter>
                <action android:name="cn.jpush.android.ui.PushActivity"/>
                <category android:name="android.intent.category.DEFAULT"/>
                <category android:name="com.vip.pinganedai"/>
            </intent-filter>
        </activity>
        <activity android:theme="@style/MyDialogStyle" android:name="cn.jpush.android.ui.PopWinActivity" android:exported="false" android:configChanges="keyboardHidden|orientation">
            <intent-filter>
                <category android:name="android.intent.category.DEFAULT"/>
                <category android:name="com.feilu.flashloan"/>
            </intent-filter>
        </activity>
        <receiver android:name="com.vip.pinganedai.hotfix.listener.MyPatchListener$ScreenBroadcastReceiver"/>
        <receiver android:name="cn.jpush.android.service.AlarmReceiver" android:exported="false"/>
        <receiver android:name="com.vip.pinganedai.jpush.JPushReceiver" android:enabled="true" android:exported="false">
            <intent-filter>
                <action android:name="cn.jpush.android.intent.REGISTRATION"/>
                <action android:name="cn.jpush.android.intent.MESSAGE_RECEIVED"/>
                <action android:name="cn.jpush.android.intent.NOTIFICATION_RECEIVED"/>
                <action android:name="cn.jpush.android.intent.NOTIFICATION_OPENED"/>
                <action android:name="cn.jpush.android.intent.CONNECTION"/>
                <category android:name="com.vip.pinganedai"/>
            </intent-filter>
        </receiver>
        <receiver android:name="com.vip.pinganedai.jpush.MyJPushMessageReceiver">
            <intent-filter>
                <action android:name="cn.jpush.android.intent.RECEIVE_MESSAGE"/>
                <category android:name="com.vip.pinganedai"/>
            </intent-filter>
        </receiver>
        <service android:name="com.talkingdata.sdk.TDAntiCheatingService" android:process="com.talkingdata.sdk.TDAntiCheatingService">
            <intent-filter>
                <action android:name="com.talkingdata.sdk.TDAntiCheatingService"/>
            </intent-filter>
        </service>
        <meta-data android:name="TD_APP_ID" android:value="0932BC785A634E6A8846BEAF82BA0840"/>
        <meta-data android:name="TD_CHANNEL_ID" android:value="Alpha"/>
        <meta-data android:name="JPUSH_CHANNEL" android:value="developer-default"/>
        <meta-data android:name="JPUSH_APPKEY" android:value="bf63abd2565f38915fe13b43"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.AuthBindCardActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.SelectRelationActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.ModifyPasswordActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.BindBankCardActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.TransactionRecordActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.ProductDetailActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.MessageActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.AgreementActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.AuthPersonInfoActivity" android:screenOrientation="portrait" android:windowSoftInputMode="adjustPan"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.AddressManngerActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.BuySuccessActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.WriteAddressActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.AddressMapActivity" android:screenOrientation="portrait"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.AuthNewPersonInfoAct"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.AuthNewAgreementAct"/>
        <activity android:name="com.vip.pinganedai.ui.usercenter.activity.RepaymentWayActivity" android:screenOrientation="portrait"/>
        <activity android:theme="@style/Udesk_Theme" android:name="cn.udesk.activity.UdeskHelperActivity" android:configChanges="keyboardHidden|navigation|orientation|uiMode|screenSize"/>
        <activity android:theme="@style/Udesk_Theme" android:name="cn.udesk.activity.UdeskHelperArticleActivity" android:configChanges="keyboardHidden|navigation|orientation|uiMode|screenSize"/>
        <activity android:theme="@style/Udesk_Theme" android:name="cn.udesk.activity.UdeskRobotActivity" android:configChanges="keyboardHidden|orientation" android:windowSoftInputMode="adjustUnspecified|stateVisible|adjustResize"/>
        <activity android:theme="@style/Udesk_Theme" android:name="cn.udesk.activity.UdeskFormActivity" android:launchMode="singleTop" android:configChanges="keyboardHidden|navigation|orientation|uiMode|screenSize"/>
        <activity android:theme="@style/Udesk_Theme" android:name="cn.udesk.activity.UdeskChatActivity" android:launchMode="singleTop" android:configChanges="keyboardHidden|navigation|orientation|uiMode|screenSize"/>
        <activity android:theme="@style/Udesk_Theme" android:name="cn.udesk.activity.UdeskZoomImageActivty" android:configChanges="keyboardHidden|navigation|orientation|uiMode|screenSize"/>
        <activity android:theme="@style/transcutestyle" android:name="cn.udesk.activity.SurvyDialogActivity" android:configChanges="keyboardHidden|navigation|orientation|uiMode|screenSize"/>
        <activity android:theme="@style/Udesk_Theme" android:name="cn.udesk.activity.OptionsAgentGroupActivity" android:configChanges="keyboardHidden|navigation|orientation|uiMode|screenSize"/>
        <activity android:theme="@style/Udesk_Theme" android:name="cn.udesk.activity.UdeskWebViewUrlAcivity" android:configChanges="keyboardHidden|navigation|orientation|uiMode|screenSize"/>
        <meta-data android:name="android.support.VERSION" android:value="25.4.0"/>
        <service android:name="com.taobao.sophix.aidl.DownloadService" android:exported="true">
            <intent-filter>
                <action android:name="com.taobao.hotfix.action"/>
                <category android:name="android.intent.category.DEFAULT"/>
            </intent-filter>
        </service>
    </application>
</manifest>
```

**Comment by sorccu** [comment]
Thanks. The same issue seems to exist in the Java project we used as a sample. I checked StringPool.cpp myself and this should now be fixed in 3.0.1.

**Comment by thinkhy** [comment]
This fix made my day!  
I hit the same problem when parse an apk which also declare the permission of JPUSH with extra atribute "android:protectionLevel" ...
